File: client/android/app/src/main/java/org/cna/keyple/demo/sale/android/nfc/slave/Application.java
/********************************************************************************
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 ********************************************************************************/
package org.cna.keyple.demo.sale.android.nfc.slave;

import org.cna.keyple.demo.sale.android.nfc.slave.di.DaggerAppComponent;
import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;
import timber.log.Timber;

public class Application extends DaggerApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        Timber.plant(new Timber.DebugTree());
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().application(this).build();
    }
}

File: client/android/app/src/main/kotlin/org/eclipse/keyple/demo/remote/Application.kt
package org.eclipse.keyple.demo.remote

import android.content.Context
import androidx.multidex.MultiDex
import dagger.android.DaggerApplication
import org.eclipse.keyple.demo.remote.di.AppComponent
import org.eclipse.keyple.demo.remote.di.DaggerAppComponent
import timber.log.Timber
import timber.log.Timber.DebugTree

class Application : DaggerApplication() {
    override fun attachBaseContext(context: Context?) {
        super.attachBaseContext(context)
        MultiDex.install(this)
    }

    override fun onCreate() {
        super.onCreate()
        Timber.plant(DebugTree())
    }

    override fun applicationInjector(): AppComponent? {
        return DaggerAppComponent.builder().application(this).build()
    }
}